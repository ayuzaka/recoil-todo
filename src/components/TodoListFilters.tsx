import { ChangeEvent, VFC } from 'react'
import { useFilterTodoList, FilterType } from '../hooks/useFilterTodoList'

type Props = {
  filter: FilterType
  onChange: (event: ChangeEvent<HTMLSelectElement>) => void
}

const Component: VFC<Props> = ({ filter, onChange }) => (
  <>
    Filter:
    <select value={filter} onChange={onChange}>
      <option value="Show All">All</option>
      <option value="Show Completed">Completed</option>
      <option value="Show Uncompleted">Uncompleted</option>
    </select>
  </>
)

const Container: VFC = () => {
  const { filter, updateFilter } = useFilterTodoList()

  const onChange = (event: ChangeEvent<HTMLSelectElement>) => {
    updateFilter(event.target.value as FilterType)
  }

  return <Component {...{ filter, onChange }} />
}

export const TodoListFilters = Container
