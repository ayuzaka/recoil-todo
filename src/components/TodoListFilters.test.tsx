import { RecoilRoot } from 'recoil'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'
import { TodoListFilters } from './TodoListFilters'

describe('初期状態', () => {
  test('render', () => {
    const { asFragment } = render(
      <RecoilRoot>
        <TodoListFilters />
      </RecoilRoot>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('フィルターの種類の初期値は「Show All」となっていること', () => {
    render(
      <RecoilRoot>
        <TodoListFilters />
      </RecoilRoot>
    )
    expect(screen.getByRole('combobox')).toHaveValue('Show All')

  })
})

describe('イベント', () => {
  beforeEach(() => {
    render(
      <RecoilRoot>
        <TodoListFilters />
      </RecoilRoot>
    )
  })

  test('フィルターの選択肢を変更できること', () => {
    const combobox = screen.getByRole('combobox')
    userEvent.selectOptions(combobox, 'Show Completed')
    expect(combobox).toHaveValue('Show Completed')
  })

  test.todo('フィルターの選択肢が変更されたとき、更新のイベントが発生すること')
})
