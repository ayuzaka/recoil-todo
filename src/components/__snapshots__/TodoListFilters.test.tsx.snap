// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`初期状態 render 1`] = `
<DocumentFragment>
  Filter:
  <select>
    <option
      value="Show All"
    >
      All
    </option>
    <option
      value="Show Completed"
    >
      Completed
    </option>
    <option
      value="Show Uncompleted"
    >
      Uncompleted
    </option>
  </select>
</DocumentFragment>
`;
